### PR TITLE
Mandatory option on questions

### LINKED ISSUE #135: Mandatory/non mandatory questions
**Which Forms version are you running?**
1.0.3

**Nextcloud or ownCloud?:**
Nextcloud

**Nextcloud/ownCloud version:** (see Nextcloud admin page)
17

### Steps to reproduce
1. create a form with non-mandatory fields
2. enter data into mandatory fields (*) leave non-mandatory fields empty

### Actual behaviour
3. cannot submit form without filling some non-mandatory fields with eg „n/a“ or some other non-null value

### Expected behaviour
Option to submit anyway or preset fields with some default text eg „n/a“

### Server configuration
**Where did you install Nextcloud from:**
Nextcloud App store 
**Operating system**:
Linux
**Web server:**
Apache2
**Database:**
MySql/MariaDB
**PHP version:**
7.2

### PR DESCRIPTION
**Pull request by jotoeri**
grafik

When editing the specific question
grafik

Closes #135

### CODE REVIEW

**Comment by skjnldsv** [comment]
Nice! :)
For the design though, we agreed on using the traditional asterisk `*` and having a `* Mandatory answer` after the description.
This is something web have been doing from the beginning of times as well as others forms app.

**Comment by skjnldsv** [comment]
Regarding the checkbox, I'm not sure. It seems to take out some readability.
What about putting it as an ActionCheckbox in the menu?

**Comment by jancborchardt** [comment]
Good stuff @jotoeri! :)

I’d agree with putting it as an action in the menu, also because on mobile it will cut off the question title too much.
Also agree with Barth’s other points

(And just a note, we generally never use italic/slanted text because it is difficult to read and not accessible.)

File: src/components/Questions/QuestionMultiple.vue [hunk 1]
```diff
@@ -45,6 +45,7 @@
 						}"
 						:name="`${id}-answer`"
 						:required="isRequired(answer.id)"
+						:aria-required="isRequired(answer.id)"
```
Comment (skjnldsv) [inline src/components/Questions/QuestionMultiple.vue:48]: Should not be needed, required is enough.
Aria is a fallback when there is not the option :)

Comment (jotoeri) [reply src/components/Questions/QuestionMultiple.vue:48]: :D so what did you want to tell me with your comment then?

Comment (skjnldsv) [reply src/components/Questions/QuestionMultiple.vue:48]: That in case you were wondering how accessibility would work, the required property is the one to go for :see_no_evil: 
https://www.youtube.com/watch?v=g9Qff0b-lHk

Comment (skjnldsv) [reply src/components/Questions/QuestionMultiple.vue:48]: Also, I did say 

> TL;DR required attribute in the inputs will be read by screen readers

Comment (jotoeri) [reply src/components/Questions/QuestionMultiple.vue:48]: :D ok.
`TL;DR` is a shortcut for...?

Comment (skjnldsv) [reply src/components/Questions/QuestionMultiple.vue:48]: "too long didn't read"  :)
https://www.urbandictionary.com/define.php?term=tl%3Bdr

**Comment by jotoeri** [comment]
Ok, so second go here - The mandatory information at description only shows up, if at least one question is marked as mandatory.

grafik

grafik

grafik

File: src/components/Questions/Question.vue [hunk 2]
```diff
@@ -107,8 +106,35 @@ export default {
...
+		 * @returns {Boolean}
+		 */
+		isMandatory: {
+			get: function() {
+				return this.mandatory
+			},
+			set: function(mandatory) {
+				this.$emit('update:mandatory', mandatory)
+			},
+		},
```
Comment (skjnldsv) [inline src/components/Questions/Question.vue:121]: I would prefer having the same syntax as we do for `onTitleChange`, enableEdit... etc
Use `:checked="this.mandatory"` and watch for the event `update:checked="onMandatoryChange"` or something :)

File: src/components/Questions/Question.vue [hunk 3]
```diff
@@ -107,8 +106,35 @@ export default {
...
+		/**
+		 * Extend text with asterisk if question is mandatory
+		 * @returns {Boolean}
+		 */
+		computedText() {
+			if (this.mandatory) {
+				return this.text + ' *'
+			} else {
+				return this.text
+			}
```
Comment (skjnldsv) [inline src/components/Questions/Question.vue:132]:
```suggestion
			}
			return this.text
```

File: src/components/Questions/Question.vue (hunk 2 above)
Comment (jotoeri) [reply src/components/Questions/Question.vue:121]: I tried, but didn't get it to work. So this was the example i found used everywhere on server^^
But seems like i found the error now on second try.^^

**Comment by jotoeri** [comment]
@jancborchardt Design of the information text is also good for you?
